llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -23615,6 +23617,23 @@ bool AArch64TargetLowering::mayBeEmittedAsTailCall(const CallInst *CI) const {
   return CI->isTailCall();
 }
 
+bool AArch64TargetLowering::isIndexingLegal(MachineInstr &MI, Register Base,
+                                            Register Offset, bool IsPre,
+                                            MachineRegisterInfo &MRI) const {
+  // HACK
+  if (IsPre)
+    return false; // Until we implement.
+
+  auto CstOffset = getIConstantVRegVal(Offset, MRI);
+  if (!CstOffset || CstOffset->isZero())
```

Comment:
Why isn't `0` a legal immediate offset here?

Reply:
Because then the G_PTR_ADD would be a NOP, and it wouldn't make sense for an pre/post-indexed load/store to add 0 to a base register.